src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -48,147 +46,95 @@
-    territoriesToAttack = new ArrayList<Territory>();
+    clearedTerritories = new ArrayList<Territory>();
     final Set<Territory> possibleTransportTerritories = new HashSet<Territory>();
-    for (final ProTerritory patd : prioritizedTerritories) {
-      territoriesToAttack.add(patd.getTerritory());
+    for (final ProTerritory patd : attackOptions) {
+      clearedTerritories.add(patd.getTerritory());
       if (!patd.getAmphibAttackMap().isEmpty()) {
         possibleTransportTerritories.addAll(data.getMap().getNeighbors(patd.getTerritory(), Matches.TerritoryIsWater));
       }
     }
-    enemyAttackOptions =
-        attackOptionsUtils.findEnemyAttackOptions(player, territoriesToAttack, new ArrayList<Territory>(
-            possibleTransportTerritories));
+    territoryManager.populateEnemyAttackOptions(clearedTerritories, new ArrayList<Territory>(
+        possibleTransportTerritories));
     territoryValueMap =
-        territoryValueUtils.findTerritoryValues(player, minCostPerHitPoint, new ArrayList<Territory>(),
-            territoriesToAttack);
-    determineTerritoriesThatCanBeHeld(prioritizedTerritories, attackMap, enemyAttackOptions, territoryValueMap);
-    removeTerritoriesThatArentWorthAttacking(prioritizedTerritories, enemyAttackOptions);
+        ProTerritoryValueUtils.findTerritoryValues(player, new ArrayList<Territory>(), clearedTerritories);
+    determineTerritoriesThatCanBeHeld(attackOptions, territoryValueMap);
+    removeTerritoriesThatArentWorthAttacking(attackOptions);
 
     // Determine how many units to attack each territory with
-    final List<Unit> alreadyMovedUnits =
-        moveOneDefenderToLandTerritoriesBorderingEnemy(attackMap, unitAttackMap, prioritizedTerritories,
-            myUnitTerritories);
-    determineUnitsToAttackWith(attackMap, enemyAttackOptions, unitAttackMap, prioritizedTerritories, transportMapList,
-        transportAttackMap, bombardMap, alreadyMovedUnits);
+    final List<Unit> alreadyMovedUnits = moveOneDefenderToLandTerritoriesBorderingEnemy(attackOptions);
+    determineUnitsToAttackWith(attackOptions, alreadyMovedUnits);
 
     // Get all transport final territories
     ProMoveUtils.calculateAmphibRoutes(player, new ArrayList<Collection<Unit>>(), new ArrayList<Route>(),
-        new ArrayList<Collection<Unit>>(), attackMap, true);
+        new ArrayList<Collection<Unit>>(), territoryManager.getAttackOptions().getTerritoryMap(), true);
 
     // Determine max enemy counter attack units and remove territories where transports are exposed
-    removeTerritoriesWhereTransportsAreExposed(attackMap, enemyAttackOptions);
+    removeTerritoriesWhereTransportsAreExposed();
 
     // Determine if capital can be held if I still own it
-    if (myCapital != null && myCapital.getOwner().equals(player)) {
-      determineIfCapitalCanBeHeld(attackMap, prioritizedTerritories, landPurchaseOptions);
+    if (ProData.myCapital != null && ProData.myCapital.getOwner().equals(player)) {
```

Comment:
Code smell here with the not null check and also the compound boolean expression. The null check can be fixed by converting the type of `myCapital` to be an `Optional`, and/or by having this boolean value be computed by ProData, eg: `boolean ProData.myCapitalIsOwnedBy(player)` (a better name would be wanted, but for example).